feature/bgp/gracefulrestart/otg_tests/exrr/exrr_test.go

```diff
@@ -0,0 +1,1872 @@
+		configuredCommunities: []struct {
+			ASNumber uint16
+			Value    uint16
+		}{
+			{errnodeprefasn, 1},
+			// TODO: stale routes are not sent to other peer, uncomment below
+			// {65535, 6},
+		},
+	},
+	{
+		prefix:            ipv6Prefix3,
+		ipType:            "ipv6",
+		bgpPeer:           ateEBGP.Name + ".BGP6.peer",
+		advertisedbgpPeer: ateIBGP.Name + ".BGP6.peer",
+		configuredCommunities: []struct {
+			ASNumber uint16
+			Value    uint16
+		}{
+			{testibgpasn, 1},
+		},
+		expectedasPath: []uint32{100, 100, 100},
+	},
+	{
+		prefix:            ipv6Prefix4,
+		ipType:            "ipv6",
+		bgpPeer:           ateIBGP.Name + ".BGP6.peer",
+		advertisedbgpPeer: ateEBGP.Name + ".BGP6.peer",
+		configuredCommunities: []struct {
+			ASNumber uint16
+			Value    uint16
+		}{
+			{noerrasn, 1},
+		},
+		expectedCommunities: []struct {
+			ASNumber uint16
+			Value    uint16
+		}{
+			{noerrasn, 1},
+		},
+	},
+	{
+		prefix:            ipv6Prefix5,
+		ipType:            "ipv6",
+		bgpPeer:           ateIBGP.Name + ".BGP6.peer",
+		advertisedbgpPeer: ateEBGP.Name + ".BGP6.peer",
+		configuredCommunities: []struct {
+			ASNumber uint16
+			Value    uint16
+		}{
+			{errnodeprefasn, 1},
+		},
+		expectedCommunities: []struct {
+			ASNumber uint16
+			Value    uint16
+		}{
+			{errnodeprefasn, 1},
+		},
+	},
+	{
+		prefix:            ipv6Prefix6,
+		ipType:            "ipv6",
+		advertisedbgpPeer: ateEBGP.Name + ".BGP6.peer",
+		configuredCommunities: []struct {
+			ASNumber uint16
+			Value    uint16
+		}{
+			{testebgpasn, 1},
+		},
+		bgpPeer:     ateIBGP.Name + ".BGP6.peer",
+		expectedmed: ptrToUint32(50),
+	},
+}
+
+func configureRoutePolicy(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	d := &oc.Root{}
+	rp := d.GetOrCreateRoutingPolicy()
+
+	for _, cfg := range communityConf {
+		communitySet := rp.GetOrCreateDefinedSets().
+			GetOrCreateBgpDefinedSets().
+			GetOrCreateCommunitySet(cfg.name)
+
+		communitySet.SetCommunityMember([]oc.RoutingPolicy_DefinedSets_BgpDefinedSets_CommunitySet_CommunityMember_Union{
+			oc.UnionString(fmt.Sprintf("%d:%d", cfg.asn, cfg.val)),
+		})
+	}
+
+	pd := rp.GetOrCreatePolicyDefinition("STALE-ROUTE-POLICY")
+	stmt10, err := pd.AppendNewStatement("10")
+	if err != nil {
+		t.Errorf("error while creating new statement %v", err)
+	}
+	stmt10.GetOrCreateActions().PolicyResult = oc.RoutingPolicy_PolicyResultType_REJECT_ROUTE
+	matchCommunitySet := stmt10.GetOrCreateConditions().GetOrCreateBgpConditions().GetOrCreateMatchCommunitySet()
+	matchCommunitySet.SetCommunitySet("NO-ERR")
+
+	stmt20, err := pd.AppendNewStatement("20")
+	if err != nil {
+		t.Errorf("error while creating new statement %v", err)
+	}
+	stmt20.GetOrCreateActions().PolicyResult = oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE
+	matchCommunitySet2 := stmt20.GetOrCreateConditions().GetOrCreateBgpConditions().GetOrCreateMatchCommunitySet()
+	matchCommunitySet2.SetCommunitySet("ERR-NO-DEPREF")
```

Comment:
Is this implementing the following from the Readme?


- identify routes tagged with community ERR-NO-DEPREF and have an action of "ACCEPT" so such routes are considered for ERR. Also ADD community STALE to the existing community list attached as part of the regular adj-rib-in post policy for the route.
- Catch-all rule to identify and accept all other prefixes, attach a local-preference of "0" and ADD community STALE to the existing community list.